postgresql/resource_postgresql_grant_test.go

```diff
@@ -600,3 +791,29 @@ func testCheckSchemaPrivileges(t *testing.T, usage, create bool) func(*terraform
 		return nil
 	}
 }
+
+func testCheckForeignDataWrapperPrivileges(t *testing.T, usage bool) func(*terraform.State) error {
+	return func(*terraform.State) error {
+		db := connectAsTestRole(t, "test_role", "postgres")
```

Comment:
```suggestion
		config := getTestConfig(t)
		dsn := config.connStr("postgres")
		
		defer func() {
                	dbExecute(t, dsn, "DROP SERVER IF EXISTS test_srv")
		}()
		db := connectAsTestRole(t, "test_role", "postgres")
```

Otherwise the `CREATE SERVER` query can fail but not as expected (permission denied), but because it already exists.
(It make me think that we could already have the same problem elsewhere and `testHasGrantForQuery` should check the error message to assert that it's a permission error)

Same for `testCheckForeignServerPrivileges`